src/Tests/Topology/MetaModel/When_parsing_string_to_connection_string.cs

```diff
@@ -233,5 +233,39 @@ public void Should_be_able_to_parse_various_domain_names(string connectionString
 
             Assert.AreEqual("namespacename", connectionString.NamespaceName);
         }
+
+        [Test]
+        [TestCase("Endpoint=sb://namespacename.servicebus.windows.net")]
+        public void Shared_access_key_should_be_optional(string connectionStringString)
+        {
+            ConnectionString connectionString;
+            ConnectionString.TryParse(connectionStringString, out connectionString);
+
+            Assert.AreEqual("namespacename", connectionString.NamespaceName);
+        }
+
+        [Test]
+        [TestCase("Endpoint=sb://namespacename.servicebus.windows.net")]
+        public void Should_equal_on_absence_of_sas_in_both(string connectionStringString)
+        {
+            ConnectionString connectionString1, connectionString2;
+            ConnectionString.TryParse(connectionStringString, out connectionString1);
+            ConnectionString.TryParse(connectionStringString, out connectionString2);
+
+            var areEqual = connectionString1.Equals(connectionString2);
+            Assert.IsTrue(areEqual);
+        }
+
+        [Test]
+        [TestCase("Endpoint=sb://namespacename.servicebus.windows.net", "Endpoint=sb://namespacename.servicebus.windows.net;SharedAccessKeyName=name;SharedAccessKey=key")]
```

Comment:
I do think using test cases here is a bit of an odd choice but I'm not against it. So approving

Reply:
Tend to agree. When I see `TestCase`, I think of permutations.
Considering that this will be ported forward, could be simplified.